tracer/test/snapshots/OpenTelemetrySdkTests.verified.txt

```diff
@@ -0,0 +1,211 @@
+﻿[
+  {
+    TraceId: Id_1,
+    SpanId: Id_2,
+    Name: Saying hello!,
+    Resource: SayHello,
+    Service: MyServiceName,
+    Type: custom,
+    Tags: {
+      attribute-bool: true,
+      attribute-boolArray: [false,true,false],
+      attribute-boolArrayEmpty: [],
+      attribute-doubleArray: [4.1,5,6],
+      attribute-doubleArrayEmpty: [],
+      attribute-intArray: [1,2,3],
+      attribute-intArrayEmpty: [],
+      attribute-string: str,
+      attribute-stringArray: ["str1","str2","str3"],
+      attribute-stringArrayEmpty: [],
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: MyServiceName,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      runtime-id: Guid_2,
+      span.kind: internal,
+      _dd.p.dm: -0
+    },
+    Metrics: {
+      attribute-double: 2.0,
+      attribute-int: 1.0,
+      process_id: 0,
+      _dd.agent_psr: 1.0,
+      _dd.top_level: 1.0,
+      _dd.tracer_kr: 1.0,
+      _sampling_priority_v1: 1.0
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_3,
+    Name: MyServiceName.internal,
+    Resource: StartSpan,
+    Service: MyServiceName,
+    Type: custom,
+    ParentId: Id_2,
+    Tags: {
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: MyServiceName,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      span.kind: internal
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_4,
+    Name: MyServiceName.internal,
+    Resource: StartSpan2,
+    Service: MyServiceName,
+    Type: custom,
+    ParentId: Id_2,
+    Tags: {
+      env: integration_tests,
+      language: dotnet,
+      otel.library.name: MyServiceName,
+      otel.status_code: STATUS_CODE_UNSET,
+      otel.trace_id: Guid_1,
+      span.kind: internal
+    }
+  },
+  {
+    TraceId: Id_1,
+    SpanId: Id_5,
+    Name: MyServiceName.internal,
+    Resource: StartRootSpan,
```

Comment:
It's weird to have a call to StartRootSpan that generates a span that isn't a root span :)

Reply:
100% agree with you Pierre! I was surprised by this as well but it looks like we're stuck with this current behavior. Hopefully that gets fixed. At the very least, we should notice when this changes in the future and can have a snapshot to match